cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java

```diff
@@ -239,4 +239,28 @@ private void removeProperty(Map<String, ConfigModel> configModel, Collection<Alt
     public boolean isEmpty() {
         return brokerConfigDiff.isEmpty();
     }
+
+    /**
+     * @param entry tested ConfigEntry
+     * @param scope scope to test
+     * @return true if the entry matches the scope
+     */
+    private boolean isScope(ConfigEntry entry, Scope scope) {
+        return configModel.get(entry.name()).getScope().equals(scope);
+    }
+
+    /**
+     * Returns configuration difference only for parameters in the specified scope
+     * @return Collection of AlterConfigOp containing difference between current and desired configuration for the specified scope
+     */
+    protected Collection<AlterConfigOp> getConfigDiff(Scope scope) {
+        Collection<AlterConfigOp> clusterWideConfigDiff = new ArrayList<>();
+        for (AlterConfigOp entry : brokerConfigDiff) {
+            if (isScope(entry.configEntry(), scope)) {
```

Comment:
Do we really need the `isScope` method? Wouldn't this be the same with lot less code?

```suggestion
            if (configModel.get(entry.name()).getScope().equals(scope))) {
```

Reply:
I was just applying the same pattern as the `isEntryReadOnly` used within `canBeUpdatedDynamically`.

Reply:
Should we just change `isEntryReadOnly()` to `isScope()` so that it is used everywhere with the given scope? e.g. `isScope(entry.configEntry(), Scope.READ_ONLY)`?

Reply:
Agree, I think leaving a method, in this case `isScope` makes the code much readable.